test/stripe.spec.ts

```diff
@@ -636,4 +636,186 @@ describe('Stripe Module', function() {
+          res.end();
+        },
+        (err, stripe, closeServer) => {
+          if (err) return done(err);
+          stripe.rawRequest(
+            'POST',
+            '/v1/customers',
+            {
+              description: 'test customer',
+              created: new Date('2009-02-13T23:31:30Z'),
+            },
+            {apiMode: 'preview', additionalHeaders: {foo: 'bar'}},
+            (err, result) => {
+              if (err) return done(err);
+              expect(result).to.deep.equal(returnedCustomer);
+              closeServer();
+              done();
+            }
+          );
+        }
+      );
+    });
+
+    it('defaults to form encoding request if not specified', (done) => {
+      return testUtils.getTestServerStripe(
+        {},
+        (req, res) => {
+          expect(req.headers['content-type']).to.equal(
+            'application/x-www-form-urlencoded'
+          );
+          const requestBody = [];
+          req.on('data', (chunks) => {
+            requestBody.push(chunks);
+          });
+          req.on('end', () => {
+            const body = Buffer.concat(requestBody).toString();
+            expect(body).to.equal(
+              'description=test%20customer&created=1234567890'
+            );
+          });
+          res.write(JSON.stringify(returnedCustomer));
+          res.end();
+        },
+        (err, stripe, closeServer) => {
+          if (err) return done(err);
+          stripe.rawRequest(
+            'POST',
+            '/v1/customers',
+            {
+              description: 'test customer',
+              created: new Date('2009-02-13T23:31:30Z'),
+            },
+            null,
+            (err, result) => {
+              if (err) return done(err);
+              expect(result).to.deep.equal(returnedCustomer);
+              closeServer();
+              done();
+            }
+          );
+        }
+      );
+    });
+
+    it('should make request with specified additional headers', (done) => {
+      return testUtils.getTestServerStripe(
+        {},
+        (req, res) => {
+          expect(req.headers.foo).to.equal('bar');
+          res.write(JSON.stringify(returnedCustomer));
+          res.end();
+        },
+        (err, stripe, closeServer) => {
+          if (err) return done(err);
+          stripe.rawRequest(
+            'GET',
+            '/v1/customers/cus_123',
+            {},
+            {additionalHeaders: {foo: 'bar'}},
+            (err, result) => {
+              if (err) return done(err);
+              expect(result).to.deep.equal(returnedCustomer);
+              closeServer();
+              done();
+            }
+          );
+        }
+      );
+    });
+
+    it('should make request successfully', async () => {
+      const response = await stripe.rawRequest('GET', '/v1/customers', {});
+
+      expect(response).to.have.property('object', 'list');
+    });
+
+    it('should allow passing in query params', (done) => {
```

Comment:
Do we have similar logic in other dynamic languages? Or sample generation definitely embeds query arguments into path.

Reply:
I can log this as a follow-up item to check that the other dynamic languages support this.